pandas/core/ops.py

```diff
@@ -1059,14 +1106,21 @@ def safe_na_op(lvalues, rvalues):
             raise
 
     def wrapper(left, right):
-
         if isinstance(right, ABCDataFrame):
             return NotImplemented
 
         left, right = _align_method_SERIES(left, right)
         res_name = get_op_result_name(left, right)
 
-        if is_datetime64_dtype(left) or is_datetime64tz_dtype(left):
+        if is_categorical_dtype(left):
+            raise TypeError("{typ} cannot perform the operation "
+                            "{op}".format(typ=type(left).__name__, op=str_rep))
```

Comment:
Not necessarily part of this PR, but this could just dispatch to the `Categorical` and it would raise `TypeError` correctly.  The exception message would be slightly changed.

Reply:
I think that dispatch_to_extension_op should subsume any is_categorial dispatching directly (but for later)